omegaconf/basecontainer.py

```diff
@@ -798,3 +815,27 @@ def _update_types(node: Node, ref_type: type, object_type: Optional[type]) -> No
         if new_ref_type is not Any:
             node._metadata.ref_type = new_ref_type
             node._metadata.optional = new_is_optional
+
+
+def _instantiate_structured_config_impl(
+    retdict: Dict[str, Any], object_type: Type[Any]
+) -> Any:
+    from ._utils import get_structured_config_data
+
+    object_type_field_names = get_structured_config_data(object_type).keys()
```

Comment:
flip the order of the two cases. in terms of importance, the regular object case is the important one.

Reply:
Done in https://github.com/omry/omegaconf/pull/502/commits/a4af7f5466cb74cbc2cde39e6f6e447a1496a992